src/tree/split_evaluator.cc

```diff
@@ -303,5 +306,195 @@ XGBOOST_REGISTER_SPLIT_EVALUATOR(MonotonicConstraint, "monotonic")
     return new MonotonicConstraint(std::move(inner));
   });
 
+/*! \brief Encapsulates the parameters required by the InteractionConstraint
+        split evaluator
+*/
+struct InteractionConstraintParams
+    : public dmlc::Parameter<InteractionConstraintParams> {
+  std::string interaction_constraints;
+  bst_uint num_feature;
+
+  DMLC_DECLARE_PARAMETER(InteractionConstraintParams) {
+    DMLC_DECLARE_FIELD(interaction_constraints)
+        .set_default("")
+        .describe("Constraints for interaction representing permitted interactions."
+                  "The constraints must be specified in the form [[0, 1], [2, 3, 4]]"
```

Comment:
in the form of nested list ... where each inner list is a group of indices of features that are allowed to interact with each other

Reply:
Fixed